src/v/cluster/members_backend.cc

```diff
@@ -302,12 +304,19 @@ ss::future<std::error_code> members_backend::reconcile() {
     }
 
     if (!_raft0->is_elected_leader()) {
+        vlog(
+          clusterlog.trace,
```

Comment:
Almost all of this change just consists in additional logging at key points of the reconciliation flow, at TRACE level. This has proven to be a place where things can hang other otherwise go wrong in large clusters.